src/webots/gui/WbGuiApplication.cpp

```diff
@@ -193,110 +194,164 @@ void WbGuiApplication::parseStreamArguments(const QString &streamArguments) {
+  parser.addOption(stderrOption);
+  QCommandLineOption logPerfomanceOption("log-performance", QCoreApplication::translate("main", "Log performance"),
+                                         QCoreApplication::translate("main", "logArgument"));
+  parser.addOption(logPerfomanceOption);
+  parser.addOptions({
+    {"disable-logging", QCoreApplication::translate("main", "Disable logging")},
+    {"enable-logging", QCoreApplication::translate("main", "Enable logging")},
+    {"log-level", QCoreApplication::translate("main", "Log level")},
+    {"no-sandbox", QCoreApplication::translate("main", "No sandbox")},
+    {"single-process", QCoreApplication::translate("main", "Single Process")},
+    {"remote-debugging-port", QCoreApplication::translate("main", "Remote Debugging Port")},
+  });
+
+  parser.process(QCoreApplication::arguments());
+  parser.parse(QCoreApplication::arguments());
+
+  QStringList args = parser.optionNames();
+
+  while (!args.isEmpty()) {
+    const QString subCommand = args.first();
+    if (subCommand == "minimize" && parser.isSet(minimizeOption)) {
+      args.pop_front();
       mShouldMinimize = true;
-    else if (arg == "--fullscreen")
+    } else if (subCommand == "fullscreen" && parser.isSet(fullscreenOption)) {
+      args.pop_front();
       mShouldStartFullscreen = true;
-    else if (arg == "--mode=stop") {
-      cout << tr("The '--mode=stop' option is deprecated. Please use '--mode=pause' instead.").toUtf8().constData() << endl;
-      mStartupMode = WbSimulationState::PAUSE;
-    } else if (arg == "--mode=pause")
-      mStartupMode = WbSimulationState::PAUSE;
-    else if (arg == "--mode=realtime")
-      mStartupMode = WbSimulationState::REALTIME;
-    else if (arg == "--mode=fast")
-      mStartupMode = WbSimulationState::FAST;
-    else if (arg == "--mode=run") {
-      cout << "Warning: `run` mode is deprecated, falling back to `fast` mode" << endl;
-      mStartupMode = WbSimulationState::FAST;
-    } else if (arg == "--no-rendering")
+    } else if (subCommand == "mode" && parser.isSet(modeOption)) {
+      args.pop_front();
+      QStringList mode = parser.values(modeOption);
+      if (mode.size() != 1) {
+        fprintf(stderr, "%s\n", qPrintable(QCoreApplication::translate("main", "Error: Must specify one mode argument.")));
+        parser.showHelp(1);
+      }
+      if (mode[0] == "stop") {
+        cout << tr("The '--mode=stop' option is deprecated. Please use '--mode=pause' instead.").toUtf8().constData() << endl;
+        mStartupMode = WbSimulationState::PAUSE;
+      } else if (mode[0] == "pause")
+        mStartupMode = WbSimulationState::PAUSE;
+      else if (mode[0] == "realtime")
+        mStartupMode = WbSimulationState::REALTIME;
+      else if (mode[0] == "fast")
+        mStartupMode = WbSimulationState::FAST;
+      else if (mode[0] == "run") {
+        cout << "Warning: `run` mode is deprecated, falling back to `fast` mode" << endl;
```

Comment:
This is not consistent with other printed message: it should probably go through `stderr` and be translated.
```suggestion
        cout << "Warning: `run` mode is deprecated, falling back to `fast` mode" << endl;
```